### ISSUE TITLE
Plot locally?

### ISSUE THREAD

**Issue by kindlychung** [ASIDataScience/scala-plotly-client#9]
Nice lib!

It would be wonderful if we can plot data locally instead of sending over internet.

**Comment by pbugnion** [comment]
I totally agree! It's on my (very long) list of things to add. If someone is willing to look into this, I'm happy to advise.

**Comment by kindlychung** [comment]
I am thinking, wouldn't this feature speed up the development of this package as well? The testing process would be much faster if it's done locally.

**Comment by pbugnion** [comment]
> The testing process would be much faster if it's done locally

If you mean that the automated tests would run faster, most of the unit tests just verify that the JSON request body is correctly formulated. The request is not sent to Plotly, so the tests already run locally.
If you mean the manual exploratory testing, the time taken to actually send the data to Plotly is such a tiny fraction of the time taken to actually think through what you want to test that the time savings are irrelevant.

Yes, the ability to plot offline is useful and is certainly something that someone needs to address, but I don't see how having it would speed up development. On the contrary, it adds an additional layer of complexity since we effectively have two different backends.

**Comment by viral612** [comment]
Curious if this is still being actively developed. Pascal - I am happy to look into building this if it helps. I am working on a project where i need plots support and having ability to programmatically store the plots will be very useful.
